pkg/scalers/gcp_storage_scaler.go

```diff
@@ -0,0 +1,230 @@
+		if err != nil {
+			return nil, err
+		}
+		meta.gcpAuthorization = auth
+	}
+
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
+		meta.metricName = kedautil.NormalizeString(fmt.Sprintf("gcp-storage-%s", val))
+	} else {
+		meta.metricName = kedautil.NormalizeString(fmt.Sprintf("gcp-storage-%s", meta.bucketName))
+	}
+
+	meta.scalerIndex = config.ScalerIndex
+	return &meta, nil
+}
+
+// IsActive checks if there are any messages in the subscription
+func (s *gcsScaler) IsActive(ctx context.Context) (bool, error) {
+	items, err := s.getItemCount(ctx, 1)
+	if err != nil {
+		return false, err
+	}
+
+	return items > 0, nil
+}
+
+func (s *gcsScaler) Close(context.Context) error {
+	if s.client != nil {
+		s.client.Close()
+	}
+	return nil
+}
+
+// GetMetricSpecForScaling returns the metric spec for the HPA
+func (s *gcsScaler) GetMetricSpecForScaling(context.Context) []v2beta2.MetricSpec {
+	targetValueQty := resource.NewQuantity(int64(s.metadata.targetObjectCount), resource.DecimalSI)
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: GenerateMetricNameWithIndex(s.metadata.scalerIndex, s.metadata.metricName),
+		},
+		Target: v2beta2.MetricTarget{
+			Type:         v2beta2.AverageValueMetricType,
+			AverageValue: targetValueQty,
+		},
+	}
+	metricSpec := v2beta2.MetricSpec{External: externalMetric, Type: externalMetricType}
+	return []v2beta2.MetricSpec{metricSpec}
+}
+
+// GetMetrics returns the number of items in the bucket (up to s.metadata.maxBucketItemsToScan)
+func (s *gcsScaler) GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
+	items, err := s.getItemCount(ctx, s.metadata.maxBucketItemsToScan)
+	if err != nil {
+		return []external_metrics.ExternalMetricValue{}, err
+	}
+
+	metric := external_metrics.ExternalMetricValue{
+		MetricName: metricName,
+		Value:      *resource.NewQuantity(int64(items), resource.DecimalSI),
+		Timestamp:  metav1.Now(),
+	}
+
+	return append([]external_metrics.ExternalMetricValue{}, metric), nil
+}
+
+// getItemCount gets the number of items in the bucket, up to maxCount
+func (s *gcsScaler) getItemCount(ctx context.Context, maxCount int) (int, error) {
+	query := &storage.Query{Prefix: ""}
+	err := query.SetAttrSelection([]string{"Name"})
+	if err != nil {
+		gcsLog.Error(err, "failed to set attribute selection")
+		return 0, err
+	}
+
+	it := s.bucket.Objects(ctx, query)
+	count := 0
+
+	for count < maxCount {
+		_, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			if strings.Contains(err.Error(), "bucket doesn't exist") {
+				gcsLog.Info("Bucket " + s.metadata.bucketName + " doesn't exist")
+				return 0, nil
+			}
+			gcsLog.Error(err, "failed to enumerate items in bucket "+s.metadata.bucketName)
+			return count, err
+		}
+		count++
+	}
+
+	gcsLog.Info(fmt.Sprintf("Counted %d items with a limit of %d", count, maxCount))
```

Comment:
```suggestion
	gcsLog.V(1).Info(fmt.Sprintf("Counted %d items with a limit of %d", count, maxCount))
```
We might want to print this under DEBUG level, to not flood logs with messages, WDYT?